tests/profiler/test_profiler.py

```diff
@@ -165,6 +167,61 @@ def test_batches_are_accessible(
         == file_list
     )
 
+    batch_request_1 = BatchRequest(
+        datasource_name="generic_csv_generator",
+        data_connector_name="daily_data_connector",
+        data_asset_name="daily_data_asset",
+        data_connector_query={
+            "index": -1,
+        },
+    )
+    # Should give most recent batch
+    validator_1 = context.get_validator(
+        batch_request=batch_request_1,
+        create_expectation_suite_with_name="my_expectation_suite_name_1",
+    )
+    metric_max = validator_1.get_metric(
+        MetricConfiguration("column.max", metric_domain_kwargs={"column": "batch_num"})
+    )
+    assert metric_max == 20
+
+    batch_request_2 = BatchRequest(
+        datasource_name="generic_csv_generator",
+        data_connector_name="daily_data_connector",
+        data_asset_name="daily_data_asset",
+        data_connector_query={
+            "index": -2,
+        },
+    )
+    validator_2 = context.get_validator(
+        batch_request=batch_request_2,
+        create_expectation_suite_with_name="my_expectation_suite_name_2",
+    )
+    metric_max = validator_2.get_metric(
+        MetricConfiguration("column.max", metric_domain_kwargs={"column": "batch_num"})
+    )
+    assert metric_max == 19
+
+    for batch_num in range(1, 20 + 1):
```

Comment:
Is 20 + 1 being written out explicitly (instead of 21) on purpose?

Reply:
Correct - just to show that it's 20 batches + 1. I should probably remove all of the magic numbers (e.g. add `total_batches: int = 20` somewhere above then use that here)

Reply:
Removed in latest commit https://github.com/great-expectations/great_expectations/pull/2697/commits/a5b9d3a2b30d729286334284ad53781c530d9178